api-js/src/affiliation/mutations/leave-organization.js

```diff
@@ -0,0 +1,358 @@
+import { t } from '@lingui/macro'
+import { GraphQLID, GraphQLNonNull } from 'graphql'
+import { fromGlobalId, mutationWithClientMutationId } from 'graphql-relay'
+
+import { leaveOrganizationUnion } from '../unions'
+
+export const leaveOrganization = new mutationWithClientMutationId({
+  name: 'LeaveOrganization',
+  description: 'This mutation allows users to leave a given organization.',
+  inputFields: () => ({
+    orgId: {
+      type: GraphQLNonNull(GraphQLID),
+      description: 'Id of the organization the user is looking to leave.',
+    },
+  }),
+  outputFields: () => ({
+    result: {
+      type: leaveOrganizationUnion,
+      description:
+        '`LeaveOrganizationUnion` resolving to either a `LeaveOrganizationResult` or `AffiliationError`.',
+      resolve: (payload) => payload,
+    },
+  }),
+  mutateAndGetPayload: async (
+    args,
+    {
+      i18n,
+      query,
+      collections,
+      transaction,
+      auth: { checkOrgOwner, userRequired, verifiedRequired },
+      loaders: { loadOrgByKey },
+      validators: { cleanseInput },
+    },
+  ) => {
+    const { id: orgKey } = fromGlobalId(cleanseInput(args.orgId))
+
+    const user = await userRequired()
+
+    verifiedRequired({ user })
+
+    const org = await loadOrgByKey.load(orgKey)
+
+    if (typeof org === 'undefined') {
+      console.warn(
+        `User: ${user._key} attempted to leave org: ${orgKey}, however it's undefined.`,
```

Comment:
I would probably say  "User ${user._key} attempted to leave undefined organization: ${orgKey}" just for the sake of clarity